### PR TITLE
Drop citations from Opinions queries if we don't have them

### PR DESCRIPTION
**Pull request by albertisfu**
As outlined in #4252, this PR introduces a feature that checks if the query contains citations. It searches for the citations in the Opinions index, and any citations that cannot be found are removed from the query. The query is then performed without the missing citations, and the user is informed accordingly.

It works for one or many missing citations in the query.

**Example with one missing citation:**
Screenshot 2024-08-27 at 10 06 58 a m

**Example with two missing citations:**
Screenshot 2024-08-27 at 10 07 16 a m

**Example with one missing citation and one that could be found.**
Screenshot 2024-08-27 at 10 07 59 a m

One remaining question:
As you can see, the modified query is shown in the search bar. Or should we display the original query instead?

### CODE REVIEW

**Comment by mlissner** [comment]
I think Google shows the way on this one and we should copy how it does typos:

image

(We don't need the "Search Instead" part though. We can do that another day.

But you can see that Google leaves the query as it was. 

For the error, I think it'd be better if it said:

> Showing results for "Le v. Aguilar Media" without the citation "XYZ." It appears we don't yet have that citation.

And I'd change the error message to use the blue warning box too (like when we find citations).

**Comment by albertisfu** [comment]
Thanks! I've tweaked it so that the original query is preserved in the search box.

- Updated the UI to use a blue alert box.
- Updated the alert message.

 Here some screenshots:

A single missing citation:
Screenshot 2024-08-27 at 3 44 44 p m

Multiple missing citations:
Screenshot 2024-08-27 at 3 43 43 p m

Mobile:
Screenshot 2024-08-27 at 3 44 15 p m

File: cl/search/templates/search.html [hunk 1]
```diff
@@ -218,6 +217,19 @@
...
         </div>
 
+        {% if search_form.type.value == SEARCH_TYPES.OPINION and missing_citations %}
+          <div class="alert alert-info" role="alert">
+            <div class="row">
+                <div class="col-xs-12">
+                  <p id="missing-citations"><i class="fa fa-info-circle fa-lg"></i>
+                    Showing results for <strong>"{{ suggested_query }}"</strong> without citation{{ missing_citations|pluralize }}
+                    {% for citation in missing_citations %} <strong>"{{ citation }}"</strong>{% if not forloop.last %}, {% endif %}{% endfor %}.</br>
+                    It appears we don't yet have that citation{{ missing_citations|pluralize }}.
```
Comment (mlissner) [inline cl/search/templates/search.html:227]: A couple tweaks:

```suggestion
                    Showing results for <strong>"{{ suggested_query }}"</strong> without citation{{ missing_citations|pluralize }}
                    {% for citation in missing_citations %} <strong>"{{ citation }}"</strong>{% if not forloop.last %}, {% endif %}{% endfor %}. It appears we don't yet have {{ missing_citations|pluralise:"that citation,those citations" }}.
```

(fixes pluralize and I don't think we need the br.)

File: cl/search/views.py [hunk 2]
```diff
@@ -667,6 +669,30 @@ def es_search(request: HttpRequest) -> HttpResponse:
...
+def remove_missing_citations(
+    missing_citations: list[FullCaseCitation], cd: CleanData
+) -> tuple[list[str], str]:
+    """Removes missing citations from the query and returns the missing
+    citations as strings and the modified query.
+
+    :param missing_citations: A list of FullCaseCitation objects representing
+    the citations that are missing from the query.
+    :param cd: A CleanData object containing the query string.
+    :return: A twp tuple containing a list of missing citation strings and the
```
Comment (mlissner) [inline cl/search/views.py:681]:
```suggestion
    :return: A two-tuple containing a list of missing citation strings and the
```

**Review by mlissner — APPROVED** [review]
A couple little things, but looks good otherwise. Thank you!

File: cl/search/templates/search.html (hunk 1 above)
Comment (albertisfu) [reply cl/search/templates/search.html]: Thanks pluralize fixed:

Screenshot 2024-08-27 at 4 36 41 p m
Screenshot 2024-08-27 at 4 35 37 p m

**Comment by mlissner** [comment]
Lovely, thank you Alberto. Set for auto-merge! Little things like this will eventually make our results really nice.

**Pull request merged**